dp/pipeline/epc_arp.c

```diff
@@ -1992,7 +1999,121 @@ epc_arp_init(void)
 #endif	/* STATIC_ARP */
 }
 
-#ifndef USE_AF_PACKET
+#ifdef USE_AF_PACKET
+static int
+data_attr_cb(const struct nlattr *attr, void *data)
+{
+	const struct nlattr **tb = data;
+	int type = mnl_attr_get_type(attr);
+
+	/* skip unsupported attribute in user-space */
+	if (mnl_attr_type_valid(attr, IFLA_MAX) < 0)
+		return MNL_CB_OK;
+
+	switch(type) {
+	case IFLA_MTU:
+		if (mnl_attr_validate(attr, MNL_TYPE_U32) < 0) {
+			perror("mnl_attr_validate");
+			return MNL_CB_ERROR;
+		}
+		break;
+	case IFLA_IFNAME:
+		if (mnl_attr_validate(attr, MNL_TYPE_STRING) < 0) {
+			perror("mnl_attr_validate");
+			return MNL_CB_ERROR;
+		}
+		break;
+	case IFLA_ADDRESS:
+		if (mnl_attr_validate(attr, MNL_TYPE_BINARY) < 0) {
+			perror("mnl_attr_validate");
+			return MNL_CB_ERROR;
+		}
+		break;
+	case IFLA_IFALIAS:
+		if (mnl_attr_validate(attr, MNL_TYPE_STRING) < 0) {
+			perror("mnl_attr_validate");
+			return MNL_CB_ERROR;
+		}
+		break;
+	}
+	tb[type] = attr;
+	return MNL_CB_OK;
+}
+
+static int
+data_cb(const struct nlmsghdr *nlh, __rte_unused void *data)
+{
+	char name[IFNAMSIZ] = "";
+	int16_t portid = -1;
+	struct nlattr *tb[IFLA_MAX+1] = {NULL};
+	struct ifinfomsg *ifm = mnl_nlmsg_get_payload(nlh);
+
+	mnl_attr_parse(nlh, sizeof(*ifm), data_attr_cb, tb);
+
+	if (tb[IFLA_IFNAME]) {
+		strcpy(name, mnl_attr_get_str(tb[IFLA_IFNAME]));
+		if (!strcmp(name, app.ul_iface_name))
+			portid = S1U_PORT_ID;
+		else if (!strcmp(name, app.dl_iface_name))
+			portid = SGI_PORT_ID;
+	}
+
+	if (portid == -1) {
+		RTE_LOG_DP(INFO, DP, "Invalid parsed port id by mnl_sock\n");
+		return MNL_CB_ERROR;
+	}
+	if (tb[IFLA_ADDRESS]) {
+		uint8_t *hwaddr = mnl_attr_get_payload(tb[IFLA_ADDRESS]);
+		if (memcmp(hwaddr, dp_ports[portid].eth_addr->addr_bytes, ETHER_ADDR_LEN) != 0 &&
+		    kni_config_mac_address(portid, hwaddr) >= 0) {
+#if DEBUG
```

Comment:
its ok to print this anyway right? No need to require a recompile. Maybe use log-level if its runtime configurable